### PR TITLE
[MM-34595] Provide a way for plugins to force the team sidebar to be visible

### PR DESCRIPTION
**Pull request by dezerb**
#### Summary
Provide a method in PluginRegistry, that forces team sidebar to be visible. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34595

#### Release Note
```release-note
Provided a method in PluginRegistry, that forces team sidebar to be visible. 
```

### CODE REVIEW

File: utils/constants.tsx [hunk 1]
```diff
@@ -307,6 +307,8 @@ export const ActionTypes = keyMirror({
     RECEIVED_PLUGIN_INSIGHT: null,
     SET_EDIT_CHANNEL_MEMBERS: null,
     NEEDS_LOGGED_IN_LIMIT_REACHED_CHECK: null,
+
+    FORCE_TEAM_SIDEBAR_TO_BE_VISIBLE: null,
```
Comment (mickmister) [inline utils/constants.tsx:311]: I'm trying to figure a better name for this, maybe `SET_TEAM_SIDEBAR_VISIBLE`? And possibly put it in a group above, but I don't see an appropriate group

File: types/store/views.ts [hunk 2]
```diff
@@ -170,6 +170,11 @@ export type ViewsState = {
         multiSelectedChannelIds: string[];
         lastSelectedChannel: string;
         firstChannelName: string;
+        forceTeamSidebarToBeVisible: boolean;
+    };
+
+    teamSidebar: {
+        forceTeamSidebarToBeVisible: boolean;
```
Comment (mickmister) [inline types/store/views.ts:177]: We only need one instance of `forceTeamSidebarToBeVisible` defined right?

File: reducers/plugins/index.ts [hunk 3]
```diff
@@ -418,11 +418,11 @@ export default combineReducers({
     // components wrapped in an object that contains an id and plugin id
     components,
 
-    // object where every key is a post type and the values are components wrapped in an
+    // object where every key is a post type and the values are components wrapped in
```
Comment (mickmister) [inline reducers/plugins/index.ts:421]: For this change and the one below, although they are helpful, it unnecessarily changes the git history of this file for the purpose of this PR. I suggest we revert this change so the history of file changes is more meaningful

File: plugins/registry.js [hunk 4]
```diff
@@ -904,4 +904,11 @@ export default class PluginRegistry {
             },
         });
     }
+
+    forceTeamSidebarToBeVisible(showSidebar) {
+        store.dispatch({
+            type: ActionTypes.FORCE_TEAM_SIDEBAR_TO_BE_VISIBLE,
+            showSidebar,
+        });
+    }
```
Comment (mickmister) [inline plugins/registry.js:913]: This is interesting because the functions on the registry are generally functions that 
"register" something, rather than just firing off an action like this. I suppose it seems fine to me. @hmhealey What are your thoughts on this?

File: components/team_sidebar/index.ts [hunk 5]
```diff
@@ -29,15 +29,19 @@ import {switchTeam, updateTeamsOrderForUser} from 'actions/team_actions.jsx';
 
 import {Preferences} from 'utils/constants';
 
+import {getForceTeamSidebarToBeVisible} from '../../selectors/views/team_sidebar';
```
Comment (mickmister) [inline components/team_sidebar/index.ts:32]: Can you place this by the other `selectors` imports above?

```suggestion
import {getForceTeamSidebarToBeVisible} from 'selectors/views/team_sidebar';
```

File: components/team_sidebar/index.ts [hunk 6]
```diff
@@ -29,15 +29,19 @@ import {switchTeam, updateTeamsOrderForUser} from 'actions/team_actions.jsx';
...
+
 import TeamSidebar from './team_sidebar';
 
 function mapStateToProps(state: GlobalState) {
     const config: Partial<ClientConfig> = getConfig(state);
 
-    const experimentalPrimaryTeam: string | undefined = config.ExperimentalPrimaryTeam;
+    const experimentalPrimaryTeam: string | undefined =
+        config.ExperimentalPrimaryTeam;
```
Comment (mickmister) [inline components/team_sidebar/index.ts:40]: I'm thinking this should stay as one line

**Review by mickmister — CHANGES_REQUESTED** [review]
Thanks for this @dezerb!! I have just one request to remove some duplication from the store types, and a few questions.

For QA manual testing, we will need a plugin that implements this, which is usually done in https://github.com/mattermost/mattermost-plugin-demo. @dezerb Would you be up for adding a piece of functionality in that project that uses this new feature? It can be implemented as a client-side slash command using [registerSlashCommandWillBePostedHook](https://developers.mattermost.com/integrate/plugins/components/webapp/reference/#registerSlashCommandWillBePostedHook)

**Comment by dezerb** [comment]
> Thanks for this @dezerb!! I have just one request to remove some duplication from the store types, and a few questions.
> 
> For QA manual testing, we will need a plugin that implements this, which is usually done in https://github.com/mattermost/mattermost-plugin-demo. @dezerb Would you be up for adding a piece of functionality in that project that uses this new feature? It can be implemented as a client-side slash command using [registerSlashCommandWillBePostedHook](https://developers.mattermost.com/integrate/plugins/components/webapp/reference/#registerSlashCommandWillBePostedHook)

Hi, I made changes according to your change request. Appropriate QA plugin changes will comes in next days.

**Review by javaguirre — COMMENTED** [review]
LGTM! I miss an e2e test to check if the new functionality (force sidebar to be visible) works?

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.
